daemon/containerd/image_delete.go

```diff
@@ -58,12 +60,24 @@ func (i *ImageService) ImageDelete(ctx context.Context, imageRef string, force,
 		return nil, err
 	}
 
+	possiblyDeletedConfigs := map[digest.Digest]struct{}{}
+	if err := i.walkPresentChildren(ctx, desc, func(d ocispec.Descriptor) {
+		if images.IsConfigType(d.MediaType) {
+			possiblyDeletedConfigs[d.Digest] = struct{}{}
+		}
+	}); err != nil {
+		return nil, err
+	}
+
 	imgID := string(desc.Digest)
 	err = i.client.ImageService().Delete(ctx, ref.String(), images.SynchronousDelete())
 	if err != nil {
 		return nil, err
 	}
 
+	// Workaround for: https://github.com/moby/buildkit/issues/3797
+	i.unleaseSnapshotsFromDeletedConfigs(ctx, possiblyDeletedConfigs)
```

Comment:
Handle the error return. It might be one of the `context.Err()` errors, signaling that the operation was not completed.